commcare_export/checkpoint.py

```diff
@@ -442,14 +448,19 @@ def __init__(
         self.since = since
         self.base_checkpoint_manager = base_checkpoint_manager
 
-    def get_since(self, checkpoint_manager):
+    def get_since(self, checkpoint_manager, data_source=None):
```

Comment:
What about using a new `PaginationMode` value instead of passing the data source here? Just thinking that in future there may be other 'cursor' based pagination APIs.